linux/vhba_hot_add_remove/vhba_device_hot_add_remove.yml

```diff
@@ -2,107 +2,106 @@
 # SPDX-License-Identifier: BSD-2-Clause
 ---
 - block:
-    - name: "Skip test case because guest OS doesn't support {{ scsi_type }} controller"
+    - include_tasks: ../utils/test_setup.yml
+      vars:
+        vmtools_check: "False"
+
+    - name: "Skip test case because guest OS doesn't support {{ new_disk_ctrl_type }} controller"
       block:
         - include_tasks: ../../common/print_test_result.yml
           vars:
             test_result: "No Run"
         - meta: end_play
       when:
-        - scsi_type in ["lsilogic", "lsilogicsas"]
+        - new_disk_ctrl_type in ["lsilogic", "lsilogicsas"]
         - guest_os_ansible_distribution in ["RedHat", "CentOS"]
         - guest_os_ansible_distribution_major_ver | int >= 8
 
-    - include_tasks: ../utils/test_setup.yml
-      vars:
-        vmtools_check: "False"
-
     - name: "Set fact of the iozone file path"
       set_fact:
         iozone_tools_path: "{{ '../../tools/iozone_64bit' if guest_os_ansible_architecture == 'x86_64' else '../../tools/iozone_32bit' }}"
     - debug:
         msg: "Will use iozone tool to test new added disk: {{ iozone_tools_path }}"
 
+    - name: "The controller type for new disk is sata controller"
+      set_fact:
+        new_ctrl_type: "scsi"
+      when: new_disk_ctrl_type in ['paravirtual', 'lsilogic', 'lsilogicsas']
+
+    - name: "The controller type for new disk is sata controller"
+      set_fact:
+        new_ctrl_type: "sata"
+      when: new_disk_ctrl_type == "sata"
+
+    - name: "The controller type for new disk is nvme controller"
+      set_fact:
+        new_ctrl_type: "nvme"
+      when: new_disk_ctrl_type == "nvme"
+
     # Get VM current disk controller info before doing hotadd testing
     - include_tasks: ../../common/vm_get_disk_controller_facts.yml
     - name: "Save original disk controller facts"
       set_fact:
-        old_scsi_facts: "{{ disk_controller_facts['disk_controller_data']['scsi'] }}"
+        disk_controller_facts_old: "{{ disk_controller_facts['disk_controller_data'][new_ctrl_type] }}"
     - name: "Get existing SCSI controller number"
       set_fact:
-        total_existing_scsi_ctrls: "{{ disk_controller_facts['disk_controller_data']['scsi'] | length }}"
+        total_existing_controller_num: "{{ disk_controller_facts['disk_controller_data'][new_ctrl_type] | length }}"
     - name: "Print existing SCSI controller number in VM '{{ vm_name }}'"
-      debug: var=total_existing_scsi_ctrls
+      debug: var=total_existing_controller_num
+
+    - block:
```

Comment:
If there is already 4 controllers with the same type, we can also do hotadd disk test?